### ISSUE TITLE
Postgres: Add support for materialized views

### ISSUE THREAD

**Issue by derickl** [sqlfluff/sqlfluff#2039]
<!--If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant.-->

## Expected Behaviour
SQLFluff should be able to parse CREATE, DROP and ALTER MATERIALIZED VIEW statements.
## Observed Behaviour
```shell
sqlfluff lint --dialect postgres -n sample.sql

== [sample.sql] FAIL                                                                                                                                                                                                                  
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | MATERIALIZED VIEW my_mat_view AS\n...'
```

```shell
sqlfluff parse --dialect postgres -n sample.sql

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        unparsable:                                           !! Expected: 'StatementSegment'
[L:  1, P:  1]      |            raw:                                              'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            raw:                                              'MATERIALIZED'
[L:  1, P: 20]      |            whitespace:                                       ' '
[L:  1, P: 21]      |            raw:                                              'VIEW'
[L:  1, P: 25]      |            whitespace:                                       ' '
[L:  1, P: 26]      |            raw:                                              'my_mat_view'
[L:  1, P: 37]      |            whitespace:                                       ' '
[L:  1, P: 38]      |            raw:                                              'AS'
[L:  1, P: 40]      |            newline:                                          '\n'
[L:  2, P:  1]      |            whitespace:                                       '  '
[L:  2, P:  3]      |            raw:                                              'SELECT'
[L:  2, P:  9]      |            newline:                                          '\n'
[L:  3, P:  1]      |            whitespace:                                       '  '
[L:  3, P:  3]      |            raw:                                              'a'
[L:  3, P:  4]      |            comma:                                            ','
[L:  3, P:  5]      |            newline:                                          '\n'
[L:  4, P:  1]      |            whitespace:                                       '  '
[L:  4, P:  3]      |            raw:                                              'b'
[L:  4, P:  4]      |            newline:                                          '\n'
[L:  5, P:  1]      |            whitespace:                                       '  '
[L:  5, P:  3]      |            raw:                                              'FROM'
[L:  5, P:  7]      |            newline:                                          '\n'
[L:  6, P:  1]      |            whitespace:                                       '      '
[L:  6, P:  7]      |            raw:                                              'my_table'
[L:  6, P: 15]      |            newline:                                          '\n'
[L:  7, P:  1]      |            whitespace:                                       '  '
[L:  7, P:  3]      |            raw:                                              'WHERE'
[L:  7, P:  8]      |            whitespace:                                       ' '
[L:  7, P:  9]      |            raw:                                              'y'
[L:  7, P: 10]      |            raw:                                              '='
[L:  7, P: 11]      |            raw:                                              "'value'"
[L:  7, P: 18]      |    statement_terminator:                                     ';'
[L:  7, P: 19]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE MATERIALIZED
                       | VIEW my_mat_view AS\n...'
```
## Steps to Reproduce

1. Create `sample.sql` using the following query
```sql
CREATE MATERIALIZED VIEW my_mat_view AS
  SELECT
  a,
  b
  FROM
      my_table
  WHERE y='value';
```
2. sqlfluff lint --dialect postgres sample.sql
3. sqlfluff parse --dialect postgres -n sample.sql

## Dialect
`postgres`

## Version
`sqlfluff, version 0.8.2 and master`
`Python 3.8.22`

## Configuration
```
N/A : Default configuration
```

**Comment by derickl** [comment]
Resolved by #2041
